src/common/transformations/src/transformations/common_optimizations/sdpa_fusion.cpp

```diff
@@ -73,37 +159,65 @@ SDPAFusion::SDPAFusion() {
             pattern_map.at(softmax).get_target_inputs().size() > 1) {
             return false;
         }
-        if (pattern_map.count(optional_add_mask) && (pattern_map.at(optional_add_mask).get_target_inputs().size() > 1 ||
-                                                     pattern_map.at(mask).get_partial_shape().size() > 4)) {
+        if (pattern_map.count(qk_opt_scaled_mask_added) &&
+            (pattern_map.at(qk_opt_scaled_mask_added).get_target_inputs().size() > 1 ||
+             pattern_map.at(mask).get_partial_shape().size() > 4)) {
             return false;
         }
 
-        Output<ov::Node> mask_value;
-        Output<ov::Node> mask_input;
-        if (pattern_map.find(optional_add_mask) != pattern_map.end()) {
-            mask_value = pattern_map.at(mask);
+        ov::Output<ov::Node> scale_node;
+        if (pattern_map.count(attn_scale) > 0) {
+            scale_node = pattern_map.at(attn_scale);
+            auto attn_scale_out_ps = scale_node.get_partial_shape();
+            if (attn_scale_out_ps.is_dynamic())
+                return false;
+            // attn_scale layer should have only single output scalar value
+            if (ov::shape_size(attn_scale_out_ps.get_shape()) != 1)
+                return false;
+            // we need to be able to cast attn_scale layer to Constant layer
+            // in order to read actual scale value
+            auto attn_scale_const_m = ov::as_type_ptr<ov::op::v0::Constant>(scale_node.get_node_shared_ptr());
+            if (!attn_scale_const_m)
+                return false;
+            auto attn_scale_val_ptr = static_cast<const void*>(attn_scale_const_m->get_data_ptr());
+            scale_node = ov::op::v0::Constant::create(T, ov::Shape{}, attn_scale_val_ptr);
```

Comment:
Something like this would be more explicit about data types:
```
auto attn_scale_val = attn_scale_const_m->cast_vector<float>();
scale_node = ov::op::v0::Constant::create(T, ov::Shape{}, attn_scale_val);
```